MediaGallerySynchronization/Model/CreateAssetFromFile.php

```diff
@@ -137,4 +145,20 @@ private function getMediaDirectory(): Read
         }
         return $this->mediaDirectory;
     }
+
+    /**
+     * Get hash image content.
+     *
+     * @param string $path
+     * @return string
+     * @throws FileSystemException
+     * @throws ValidatorException
+     */
+    private function getHashImageContent(string $path): string
+    {
+        $mediaDirectory = $this->filesystem->getDirectoryRead(DirectoryList::MEDIA);
+        $imageContent = $mediaDirectory->readFile($this->getRelativePath($path));
+        $hashedImageContent = sha1($imageContent);
```

Comment:
Could you please wrap the `sha1` hashing call to a new class to be able to reuse it.

`Magento/MediaGallerySynchronization/Model/GetContentHash` class implementing `Magento/MediaGallerySynchronizationApi/Model/GetContentHashInterface` with a single `public function execute(string $content): string` returning the hash.

Reply:
Why cannot we use `sha1_file` here? Another point I would consider is moving this "anti corruption layer" layer to framework.

Reply:
@sivaschenko this is noted.
@lenaorobei I’ve looked up about `sha1` and `sha1_file` earlier and I think `sha1_file` should be used instead since we’re dealing with image files.